game-app/game-core/src/main/java/org/triplea/sound/ClipPlayer.java

```diff
@@ -129,14 +126,18 @@ public class ClipPlayer {
     }
   }
 
-  protected final Map<String, List<URL>> sounds = new HashMap<>();
+  private final Map<String, List<URL>> sounds = new ConcurrentHashMap<>();
```

Comment:
Why `ConcurrentHashMap`? Asked another way, what/where is the concurrency concern?

Reply:
The `play` method caches sound paths in this map. In theory `play` could be called from multiple threads. I doubt that ever happens (because in this case this should have went wrong in production already), so I wanted to make sure this would never happen

Reply:
Even if called from multiple threads, it wouldn't be a problem unless we mutate the contents of the map, right?

Reply:
Not sure about that to be honest. I changed the code to use `computeIfAbsent`. So it's clear that individual entries are never modified. The map as a whole however is, not sure how robust the implementation is there